// ng2-slim-loading-bar/index.js
// Copyright (C) 2016 Sergey Akopkokhyants
// This project is licensed under the terms of the MIT license.
// https://github.com/akserg/ng2-slim-loading-bar
'use strict';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var slim_loading_bar_component_1 = require('./src/slim-loading-bar.component');
var slim_loading_bar_service_1 = require('./src/slim-loading-bar.service');
__export(require('./src/slim-loading-bar.component'));
__export(require('./src/slim-loading-bar.service'));
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = {
    providers: [slim_loading_bar_service_1.SlimLoadingBarService],
    directives: [slim_loading_bar_component_1.SlimLoadingBarComponent]
};
var SlimLoadingBarModule = (function () {
    function SlimLoadingBarModule() {
    }
    SlimLoadingBarModule.forRoot = function () {
        return {
            ngModule: SlimLoadingBarModule,
            providers: [slim_loading_bar_service_1.SlimLoadingBarService]
        };
    };
    SlimLoadingBarModule = __decorate([
        core_1.NgModule({
            imports: [common_1.CommonModule],
            declarations: [slim_loading_bar_component_1.SlimLoadingBarComponent],
            exports: [common_1.CommonModule, slim_loading_bar_component_1.SlimLoadingBarComponent]
        }), 
        __metadata('design:paramtypes', [])
    ], SlimLoadingBarModule);
    return SlimLoadingBarModule;
}());
exports.SlimLoadingBarModule = SlimLoadingBarModule;
